In [28]:
import numpy as np 
import pandas as pd 

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/titanic/train.csv
/kaggle/input/titanic/test.csv
/kaggle/input/titanic/gender_submission.csv


In [29]:
train_data = pd.read_csv('/kaggle/input/titanic/train.csv')
train_data

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

Aqui podemos ver que temos um total de 890 linhas, e que possuímos as colunas:
1. PassengerId: ID do passageiro
2. Survived: 0 = não sobreviveu, 1 = sobreviveu
3. Pclass: Classe do passageiro
4. Name: Nome do passgeiro
5. Sexo: Gênero do passgeiro
6. Age: Idade do passageiro
7. SibSp: Total de irmãos/esposas do passageiro à bordo
8. Parch: Total de pais/filhos do passageiro à bordo
9. Ticket: N° da passagem
10. Fare: Preço da passagem
11. Cabin: N° da cabine
12. Embarked: Cidade em que o passageiro embarcou (C=Cherbourg, Q= Queenstown, S= Southampton)

In [30]:
#Agora vamos checar se há dados nulos
train_data.isnull().sum().sort_values(ascending=False)

Cabin          687
Age            177
Embarked         2
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
SibSp            0
Parch            0
Ticket           0
Fare             0
dtype: int64

Assim vemos que existem valores nulos nas colunas "Cabin", "Age" e "Embarked". E agora faremos uma limpa nos dados

In [31]:
#Como a maioria dos dados da coluna "Cabin" são nulos, iremos apagá-la
train_data.drop("Cabin", axis='columns', inplace=True)

#E removeremos algumas linhas da coluna "Embarked"
train_data.dropna(subset=['Embarked'], inplace=True)

train_data.isnull().sum().sort_values(ascending=False)

Age            177
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Embarked         0
dtype: int64

In [32]:
mulheres = train_data.loc[train_data.Sex == 'female']['Survived']
mulheres_sobreviventes = sum(mulheres)/len(mulheres)

print("porcentagem de mulheres sobreviventes: ", mulheres_sobreviventes*100,"%")

homens = train_data.loc[train_data.Sex == 'male']['Survived']
homens_sobreviventes = sum(homens)/len(homens)

print("porcentagem de homens sobreviventes: ", homens_sobreviventes*100, "%")

porcentagem de mulheres sobreviventes:  74.03846153846155 %
porcentagem de homens sobreviventes:  18.890814558058924 %


Com as análises acima podemos ver que a maioria dos sobreviventes foram mulheres